crates/module_system/message_api/src/handler.rs

```diff
@@ -1,24 +1,53 @@
 //! The raw handler and host backend interfaces.
-use crate::code::ErrorCode;
+use crate::code::{ErrorCode, SystemCode};
 use crate::packet::MessagePacket;
 
 /// A handler for an account.
 pub trait RawHandler {
-    /// Handle a message packet.
-    fn handle(
+    /// Handle a message.
+    fn handle_msg(
+        &self,
+        message_packet: &mut MessagePacket,
+        callbacks: &mut dyn HostBackend,
+        allocator: &dyn Allocator,
+    ) -> Result<(), ErrorCode> {
+        Err(ErrorCode::SystemCode(SystemCode::MessageNotHandled))
+    }
+
+    /// Handle a query message.
+    fn handle_query(
         &self,
         message_packet: &mut MessagePacket,
         callbacks: &dyn HostBackend,
         allocator: &dyn Allocator,
-    ) -> Result<(), ErrorCode>;
+    ) -> Result<(), ErrorCode> {
+        Err(ErrorCode::SystemCode(SystemCode::MessageNotHandled))
+    }
+
+    /// Handle a system message.
+    fn handle_system(
+        &self,
+        message_packet: &mut MessagePacket,
+        callbacks: &mut dyn HostBackend,
+        allocator: &dyn Allocator,
+    ) -> Result<(), ErrorCode> {
+        Err(ErrorCode::SystemCode(SystemCode::MessageNotHandled))
```

Comment:
is this a todo or on purpose?

Reply:
it's on purpose but probably not necesary

Reply:
actually it is used. these are method stubs so implementors don't need to implement them all